src/components/vr_cooperation/src/command/base_json_request.cc

```diff
@@ -154,8 +132,12 @@ void BaseJsonRequest::SendMessageToHMI(
 void BaseJsonRequest::SendMessageToMobile(
     application_manager::MessagePtr message) {
   LOG4CXX_AUTO_TRACE(logger_);
-  PrepareRequestMessageForMobile(gpb_message_.rpc(), gpb_message_.params(),
-                                 message);
+
+  message->set_protocol_version(application_manager::ProtocolVersion::kV2);
+  message->set_correlation_id(gpb_message_.correlation_id());
+  message->set_function_id(GetMobileFunctionID(gpb_message_.rpc()));
+
+  message->set_message_type(application_manager::MessageType::kRequest);
```

Comment:
actually I don't think we need separated function just for prepare common part, we can put common part in SendMsgToMobile of Base class directly.